src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -594,6 +594,14 @@ static int do_map()
   }
 
   size = info.size;
+
+  if (size > (1UL << 32) * 512) {
+    r = -EINVAL;
```

Comment:
I don't think ```E2BIG``` is a correct error code for the reason Mykola stated. You might be able to get away with ```EFBIG``` for "file too large"